lib/pbench/server/api/resources/upload_api.py

```diff
@@ -51,69 +51,75 @@ def __init__(self, config, logger):
+            )
+        filename = secure_filename(filename_hdr)
+        if not self.allowed_file(filename):
+            self.logger.warning(
+                f"Tarfile upload: Bad file extension received for file {filename}"
             )
-        filename = secure_filename(request.headers.get("filename"))
+            abort(400, message="File extension not supported. Only .xz")
 
-        if not request.headers.get("Content-MD5"):
-            self.logger.debug(
-                f"Tarfile upload: Post operation failed due to missing md5sum header for file {filename}"
+        md5sum = request.headers.get("Content-MD5")
+        if not md5sum:
+            self.logger.warning(
+                "Tarfile upload: PUT operation failed due to missing md5sum"
+                f" header for file {filename}"
             )
             abort(
                 400,
-                message="Missing md5sum header, POST operation requires md5sum of an uploaded file in header",
+                message="Missing md5sum header, PUT operation requires md5sum"
+                " of an uploaded file in header",
             )
-        md5sum = request.headers.get("Content-MD5")
 
-        if not self.allowed_file(filename):
-            self.logger.debug(
-                f"Tarfile upload: Bad file extension received for file {filename}"
+        content_length_hdr = request.headers.get("Content-Length")
+        if not content_length_hdr:
+            self.logger.warning(
+                "Tarfile upload: No Content-Length header value found for"
+                f" file {filename}"
             )
-            abort(400, message="File extension not supported. Only .xz")
-
+            abort(400, message="Missing required content-length header")
         try:
-            content_length = int(request.headers.get("Content-Length"))
+            content_length = int(content_length_hdr)
         except ValueError:
-            self.logger.debug(
-                f"Tarfile upload: Invalid content-length header, not an integer for file {filename}"
+            self.logger.warning(
+                "Tarfile upload: Invalid content-length header, not an"
+                f" integer for file {filename}"
             )
             abort(400, message="Invalid content-length header, not an integer")
-        except Exception:
-            self.logger.debug(
-                f"Tarfile upload: No Content-Length header value found for file {filename}"
+        if content_length > self.max_content_length:
+            self.logger.warning(
+                "Tarfile upload: Content-Length exceeded maximum upload size"
+                f" allowed fo file {filename}"
+            )
+            abort(
+                400,
+                message=f"Payload body too large, {content_length:d} bytes,"
+                " maximum size should be less than or equal to"
+                f" {humanize.naturalsize(self.max_content_length)}",
```

Comment:
This message is botched.  How about, "Upload size ({content_length:d} bytes) must be no more than {humanize.naturalsize(self.max_content_length)}."